test/user.test.js

```diff
@@ -300,6 +338,20 @@ describe('User', function() {
       });
     });
 
+    // TODO (pradnya)- Throws "Unhandled rejection Error: login failed". Cannot call method 'then' of undefined
+    it.skip('Login should only allow correct credentials - promise variant', function(done) {
+      User.login(invalidCredentials)
+        .then(function(accessToken) {
+          assert(!accessToken);
+          done();
+        })
+        .catch(function(err) {
+          assert(err);
+          assert.equal(err.code, 'LOGIN_FAILED');
+          done();
+        });
+    });
+
```

Comment:
Rather than falling into `catch`, this `User.login` throws "Unhandled rejection Error: login failed". And so, when executing the test it says "Cannot call method 'then' of undefined". Whereas, https://github.com/strongloop/loopback/pull/1493/files#diff-5d81af95449574544e7317f3eda2badeR259, correctly returns `fn.promise`